Index: modules/lua/mod_lua.c
===================================================================
--- modules/lua/mod_lua.c	(revision 1668826)
+++ modules/lua/mod_lua.c	(working copy)
@@ -1072,7 +1072,11 @@
         else {
             luaL_Buffer b;
             luaL_buffinit(lvm, &b);
+#if LUA_VERSION_NUM >= 503
+            lua_dump(lvm, ldump_writer, &b, 0);
+#else
             lua_dump(lvm, ldump_writer, &b);
+#endif
             luaL_pushresult(&b);
             spec->bytecode_len = lua_strlen(lvm, -1);
             spec->bytecode = apr_pstrmemdup(cmd->pool, lua_tostring(lvm, -1),
